src/client/content_mapblock.cpp

```diff
@@ -958,6 +973,27 @@ void MapblockMeshGenerator::drawPlantlikeQuad(float rotation, float quad_offset,
 		vertex.rotateXZBy(rotation + rotate_degree);
 		vertex += offset;
 	}
+	
+	// rotate when facedir is used
+	if (facedir) {
+	  int axisdir = facedir >> 2;
+		facedir &= 0x03;
+		for (v3f &vertex : vertices) {
+			switch(facedir) {
+				case 1: vertex.rotateXZBy( -90); break;
+				case 2: vertex.rotateXZBy( 180); break;
+				case 3: vertex.rotateXZBy( 90); break;
+			}
+			switch(axisdir) {
+				case 1: vertex.rotateYZBy( 90); break;
+				case 2: vertex.rotateYZBy( -90); break;
+				case 3: vertex.rotateXYBy( -90); break;
+				case 4: vertex.rotateXYBy( 90); break;
+				case 5: vertex.rotateXYBy( -180); break;
+			}
+		}
+	}
+	
```

Comment:
You can optimize this a bit by reusing the existing rotation, like so:
```cpp
	switch (facedir & 0x03) {
		case 1: rotation -= 90; break;
		case 2: rotation += 180; break;
		case 3: rotation += 90; break;
	}

	for (v3f &vertex : vertices) {
		vertex.rotateXZBy(rotation + rotate_degree);
		vertex += offset;
	}
	
	if (int axisdir = facedir >> 2) {
		for (v3f &vertex : vertices) {
			switch (axisdir) {
				case 1: vertex.rotateYZBy( 90); break;
				case 2: vertex.rotateYZBy(-90); break;
				case 3: vertex.rotateXYBy(-90); break;
				case 4: vertex.rotateXYBy( 90); break;
				case 5: vertex.rotateXYBy(-180); break;
			}
		}
	}
```

Reply:
I have delivered this part from code in file [mapnode.cpp.](https://github.com/minetest/minetest/blob/90a7bd6a0afb6509e96bcb373f95b448ee7f3b1d/src/mapnode.cpp#L260) transformNodeBox function.

```cpp
        		int facedir = n.getFaceDir(nodemgr, true);
        		u8 axisdir = facedir&gt;&gt;2;
        		facedir&amp;=0x03;
        		for (aabb3f box : fixed) {
        			if (nodebox.type == NODEBOX_LEVELED)
        				box.MaxEdge.Y = (-0.5f + n.getLevel(nodemgr) / 64.0f) * BS;
      
        			switch (axisdir) {
        			case 0:
        				if(facedir == 1)
        				{
        					box.MinEdge.rotateXZBy(-90);
        					box.MaxEdge.rotateXZBy(-90);
        				}
        				else if(facedir == 2)
        				{
        					box.MinEdge.rotateXZBy(180);
        					box.MaxEdge.rotateXZBy(180);
        				}
        				else if(facedir == 3)
        				{
        					box.MinEdge.rotateXZBy(90);
        					box.MaxEdge.rotateXZBy(90);
        				}
        				break;
        			case 1: // z+
        				box.MinEdge.rotateYZBy(90);
        				box.MaxEdge.rotateYZBy(90);
        				<span class="pl-k">if</span>(facedir == <span class="pl-c1">1</span>)
```
So, isn't it be better to use the switch with if for optimization?

Reply:
The code I suggested should be fine (and also the nodebox code does not directly transfer to plantlike)

Reply:
Ok. I understand. Fixed.